crates/control/src/localization_recorder.rs

```diff
@@ -0,0 +1,157 @@
+use std::{
+    collections::{BTreeMap, HashSet},
+    fs::File,
+    io::Write,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use bincode::serialize;
+use color_eyre::{eyre::Context, Result};
+use context_attribute::context;
+use framework::{HistoricInput, PerceptionInput};
+use nalgebra::Isometry2;
+use serde::{Deserialize, Serialize};
+use types::{GameControllerState, LineData, PrimaryState};
+
+pub struct LocalizationRecorder {
+    recording: Option<File>,
```

Comment:
We should use a buffered writer here. e.g. `std::io::BufWriter`

Reply:
Done